### PR TITLE
support alter audit policy in oracle

### PR DESCRIPTION
**Pull request by Trydamere**
Ref https://github.com/apache/shardingsphere/issues/6480 & https://github.com/apache/shardingsphere/issues/18431

Changes proposed in this pull request:
- Add alter audit policy grammar
- Support alter audit policy in Oracle
- Add test

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 1]
```diff
@@ -2548,3 +2548,184 @@ resolveClauses
 resolveClause
     : LP_ matchString DOT_? (schemaName | MINUS_) RP_
     ;
+
+alterAuditPolicy
+    : ALTER AUDIT POLICY policyName
+        (ADD (privilegeAuditClause)? (actionAuditClause)? (roleAuditClause)? (ONLY TOPLEVEL)?)?
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:2554]: Can you extract these rules to new sub syntax rules?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 2]
```diff
@@ -2548,3 +2548,184 @@ resolveClauses
...
+privilegeAuditClause
+    : PRIVILEGES systemPrivilegeClause (COMMA_ systemPrivilegeClause)*
+    ;
+
+actionAuditClause
+    : (standardActions | componentActions)*
+    ;
+
+standardActions
+    : ACTIONS (objectAction ON (DIRECTORY directoryName | MINING MODEL objectName | objectName) | systemAction)
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:2568]: Can you extract these rules to new sub syntax rules?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 (hunk 2 above)
Comment (Trydamere) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:2568]: ok

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/AlterAuditPolicyStatement.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+package org.apache.shardingsphere.sql.parser.sql.common.statement.ddl;
+
+import lombok.ToString;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.AbstractSQLStatement;
+
+/**
+ * Alter audit policy statement.
+ */
+@ToString
```
[the remark this answers is not in the document]
Comment (Trydamere) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/ddl/AlterAuditPolicyStatement.java:26]: ok

**Review by strongduanmu — APPROVED** [review]
@Trydamere Thank you for your great contribution, merged.

**Pull request merged**